src/typetree.jl

```diff
@@ -110,6 +110,9 @@ function typetree(@nospecialize(T), ctx, dl)
     end
 
     if fieldcount(T) == 0
+        if T <: Function
```

Comment:
If the Field count is 0 we might want to just return `TypeTree()` since it is a ghost or singleton?